compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1761,6 +1758,20 @@ object Parsers {
       if in.isNewLine then in.nextToken()
       tdef
 
+    /** Quoted ::=  ‘'’ ‘{’ Block ‘}’
+     *           |  ‘'’ ‘[’ Type ‘]’
```

Comment:
Couldn't the grammar be refactored so that we can drop this case and only have Block and TypeBlock?

Reply:
This should just be `TypeBlock`.  `Type` is already one of the cases of `TypeBlock`.

In `syntax.md` we have the correct one

```
Quoted            ::=  ‘'’ ‘{’ Block ‘}’
                    |  ‘'’ ‘[’ TypeBlock ‘]’
```